onert-micro/onert-micro/src/train/tests/BostonHousingTask.test.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "train/tests/boston_housing_task/BostonHousingTask.h"
+#include "train/tests/OMTestUtils.h"
+
+#include <gtest/gtest.h>
+
+namespace onert_micro
+{
+namespace train
+{
+namespace test
+{
+
+using namespace testing;
+
+class BostonHousingTaskTest : public ::testing::Test
+{
+public:
+  BostonHousingTaskTest()
+  {
+    // Set user defined training settings
+    const uint32_t training_epochs = 30;
+    const float lambda = 0.1f;
+    const uint32_t batch_size = 32;
+    // Train all layers
+    const uint32_t num_train_layers = 0;
+    const onert_micro::OMLoss loss = onert_micro::MSE;
+    const onert_micro::OMTrainOptimizer train_optim = onert_micro::ADAM;
+    const float beta = 0.9;
+    const float beta_squares = 0.999;
+    const float epsilon = 1e-07;
+
+    config.train_mode = true;
+    {
+      onert_micro::OMTrainingContext train_context;
+      train_context.batch_size = batch_size;
+      train_context.num_of_train_layers = num_train_layers;
+      train_context.learning_rate = lambda;
+      train_context.loss = loss;
+      train_context.optimizer = train_optim;
+      train_context.beta = beta;
+      train_context.beta_squares = beta_squares;
+      train_context.epsilon = epsilon;
+      train_context.epochs = training_epochs;
```

Comment:
train_context.num_epoch ?

Reply:
train_context.num_epoch by default before training is zero. Maybe it is bad naming: `epochs` means  the number of epochs, and `num_epoch` means the value of the current epoch during training.